Adds migration to update CommunicationRequest payload[0].contentString for a subset...

mcjustin opened the pull request:
Adds migration to update CommunicationRequest payload[0].contentString for a subset of resources to a new message, per request from the research team ("Request to update outgoing message").

```
•	Current text: Thinking of you, {name} 🙂 I'm not always available but if you need to talk to someone, the National Hotline number is 988 & press 1 and the Veterans Crisis Line number is 838-255. -{username}
•	New text: Thinking of you, {name} 🙂 I'm not always available but if you need to talk to someone, you can try Vets4Warriors or ObjectiveZero for peer chat/support/advice for veterans (and their families/loved ones). Here are their links: [https://vets4warriors.com/talk-to-us/](https://urldefense.com/v3/__https:/vets4warriors.com/talk-to-us/__;!!K-Hz7m0Vt54!hhFDG27O-rJlLiPdK6PIjegwG2MbsXEDZLrZ0z44KUFz-q4HlGKDcfu8yztDhULS8YlhNPmo1_PaBA$) and [https://www.objectivezero.org/veterans](https://urldefense.com/v3/__https:/www.objectivezero.org/veterans__;!!K-Hz7m0Vt54!hhFDG27O-rJlLiPdK6PIjegwG2MbsXEDZLrZ0z44KUFz-q4HlGKDcfu8yztDhULS8YlhNPmJpo5AKg$). For immediate help, the National Hotline number is 988 & press 1 and the Veterans Crisis Line text number is 838-255. -{username}
```

See corresponding change to the PlanDefinition: https://github.com/uwcirg/isacc-messaging-client-sof/pull/102

Note to self: I used Cursor for this.

pbugni committed:
Merge pull request #80 from uwcirg/develop

ISACC.25.02.11 dev->main

mcjustin committed:
Adds migration to update payload[0].contentString for a subset of
CommunicationRequest resources to a new message, per request from
the research team ("Request to update outgoing message").

mcjustin commented:
@ivan-c  I expect this will be a hotfix, but I don't know how migrations work. Should I switch this to merge w/ `main`?

mcjustin commented:
> @ivan-c I expect this will be a hotfix, but I don't know how migrations work. Should I switch this to merge w/ `main`?

**_@ivan-c  & @pbugni  can you please review this week?_** 

I was ignoring this one because funding was in question for ISACC, but that's been resolved and I'd like to get this out next week, likely with another migration that I'll write.

mcjustin commented:
> > @ivan-c I expect this will be a hotfix, but I don't know how migrations work. Should I switch this to merge w/ `main`?
> 
> **_@ivan-c & @pbugni can you please review this week?_**
> 
> I was ignoring this one because funding was in question for ISACC, but that's been resolved and I'd like to get this out next week, likely with another migration that I'll write.

The migration seems to have run properly at isacc.dev. I will likely have one more to do early next week, will do that on top of this since it will refer to this migration ID, so I'll switch this back to 'draft' state.

pbugni commented:
@mcjustin the query seems questionable to me, and in contrast to the description above.  I believe this is updating the content for ALL "active" CommunicationRequests.

Is that the intent? 

Also, we should be paging through results, as `_count` could be ignored and at some point, might get exceeded.

mcjustin committed:
Clarify downgrade output ("no action taken").
Change alembic `revision` to a UUIDv4 string.
Increase search _count parameter from 1000 to 5000. 1000 is safe as
there are only 696 CommunicationRequest resources in the system with
status = 'active', but ack no need to cut it close. Paging would
certainly be more robust, so don't use this as a pattern for other
systems.

mcjustin commented:
> @mcjustin the query seems questionable to me, and in contrast to the description above. I believe this is updating the content for ALL "active" CommunicationRequests.
> 
> Is that the intent?
> 
> Also, we should be paging through results, as `_count` could be ignored and at some point, might get exceeded.

Thanks @pbugni 
- The initial query does indeed include all active CR's (n = 696 on prod atm), but when the script iterates over the results it only acts on those matching the undesirable text (n=122).
- I've just increased the count from 1000 to 5000; even tho it will only match 696 today, no sense in cutting it close. 
- I agree re paging being more robust, but this is a one-off and when I ran it on dev on 2025-10-31 it made the change to the desired records (n=6 there).

mcjustin commented:
@ivan-c  I've marked this as ready for review. I don't need to add another migration to this after all, so it's good to go. Per above, I did test on [dev](https://femr.isacc.dev.cirg.uw.edu/). I see that prod was a little tricky at the last deploy in February (per [slack](https://cirg.slack.com/archives/C03HTRD8RRS/p1739328365448799?thread_ts=1739230144.310849&cid=C03HTRD8RRS)), so I'll defer eval & prod updates to you.

Is this the path forward? If so, do you want me to go through creating the tag & release?
1. Merge this (to `develop`).
2. Merge `develop` into `main`.
3. Create the tag & release off `main`.
4. Switch [eval](https://femr.isacc.eval.cirg.uw.edu) to the tag and redeploy.
5. Justin: review eval, compare with `seaborgium:/home/mcjustin/CommunicationRequest.isacc-eval.active.gt2025-11-11.json`
6. Switch prod to the tag and deploy. 
8. Review prod, compare with `seaborgium:/home/mcjustin/CommunicationRequest.isacc-cove-prod.active.gt2025-11-11.json`
